### PR TITLE
(#3608) - clarify filtered replication/changes

### PR DESCRIPTION
**Pull request by nolanlawson**
Filtered replication is way too complicated. Our goal is consistency with CouchDB, though, so I'm glad we implemented all these different options.

But golly, there are a lot.

### CODE REVIEW

File: docs/_includes/api/replication.html [hunk 1]
```diff
@@ -14,11 +14,17 @@
 
 * `options.live`: If `true`, starts subscribing to future changes in the `source` database and continue replicating them.
 * `options.retry`: If `true` will attempt to retry replications in the case of failure (due to being offline), using a backoff algorithm that retries at longer and longer intervals until a connection is re-established. Only applicable if `options.live` is also `true`.
-* `options.filter`: Reference a filter function from a design document to selectively get updates.
-* `options.query_params`: Query params sent to the filter function.
-* `options.doc_ids`: Only replicate docs with these ids (array of strings).
 * `options.since`: Replicate changes after the given sequence number.
-* `options.create_target`: Create target database if it does not exist. Only for server replications.
```
Comment (nolanlawson) [inline docs/_includes/api/replication.html]: `create_target` is not needed anymore, because we don't have server-initiated replication anymore

**Comment by nolanlawson** [comment]
Closing #3616 because I can handle that in this PR.

**Comment by nolanlawson** [comment]
Updated to include info about filtering deleted docs.

**Comment by daleharvey** [comment]
Doc updates look great, thanks

https://github.com/pouchdb/pouchdb/commit/22c6b4be03b9a50792bee13f380952f335e899fc

File: docs/_includes/api/replication.html (hunk 1 above)
Comment (gr2m) [inline docs/_includes/api/replication.html]: Hey Nolan, I totally missed that `create_target` has been removed. Does that mean that if I replicate from `db1` to `db2`, `db2` will now always be created if it does not exist yet? Or does `create_target` still have an effect for that? We use the option here: https://github.com/hoodiehq/pouchdb-hoodie-sync/blob/master/lib/connect.js

**Pull request closed without merging**